File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Foundation/SynchronizedHashtable4.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Foundation;

namespace Db4objects.Db4o.Foundation
{
	/// <exclude></exclude>
	public class SynchronizedHashtable4 : IDeepClone
	{
		private readonly Hashtable4 _delegate;

		private SynchronizedHashtable4(Hashtable4 delegate_)
		{
			_delegate = delegate_;
		}

		public SynchronizedHashtable4(int size) : this(new Hashtable4(size))
		{
		}

		public virtual object DeepClone(object obj)
		{
			lock (this)
			{
				return new Db4objects.Db4o.Foundation.SynchronizedHashtable4((Hashtable4)_delegate
					.DeepClone(obj));
			}
		}

		public virtual void Put(object key, object value)
		{
			lock (this)
			{
				_delegate.Put(key, value);
			}
		}

		public virtual object Get(object key)
		{
			lock (this)
			{
				return _delegate.Get(key);
			}
		}
	}
}

File: db4oj/core/src/com/db4o/foundation/SynchronizedHashtable4.java
/* Copyright (C) 2007  Versant Inc.  http://www.db4o.com */

package com.db4o.foundation;


/**
 * @exclude
 */
public class SynchronizedHashtable4 implements DeepClone {
    
    private final Hashtable4 _delegate;
    
    private SynchronizedHashtable4(Hashtable4 delegate_){
        _delegate = delegate_;
    }

    public SynchronizedHashtable4(int size) {
        this(new Hashtable4(size));
    }

    public synchronized Object deepClone(Object obj) {
        return new SynchronizedHashtable4((Hashtable4)_delegate.deepClone(obj));
    }

    public synchronized void put(Object key, Object value) {
        _delegate.put(key, value);
    }

    public synchronized Object get(Object key) {
        return _delegate.get(key);
    }

}
